credentials/credentials.go

```diff
@@ -77,23 +111,82 @@ func GetNewRelicLicenseKey(ctx context.Context, conf *config.Configuration) (str
 		return conf.LicenseKey, nil
 	}
 
+	var err error
+	var licenseKey string
+
 	secretId := getLicenseKeySecretId(conf)
+	if secretId != "" {
+		licenseKey, err = tryLicenseKeyFromSecret(ctx, secretId)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	parameterName := getLicenseKeyParameterName(conf)
+	if parameterName != "" {
+		licenseKey, err = tryLicenseKeyFromParameter(ctx, parameterName)
+		if err == nil {
+			return licenseKey, nil
+		}
+	}
+
+	envLicenseKey, found := os.LookupEnv(defaultSecretId)
+	if found {
+		return envLicenseKey, nil
+	}
+
+	// Check for errors while fetching from Secrets Manager or SSM Parameter Store
+	if err != nil {
+		return "", err
+	}
```

Comment:
This is a really weird error check. The flow control in this function is really quite confusing.

Reply:
I was trying to handle the case where both Secrets Manager or Parameter Store were configured. Hopefully the new version is more straightforward.